### PR TITLE
Add feature flag for License-based Checkout, Milestone 1

### PR DESCRIPTION
**Pull request by elliottprogrammer**
#### Changes proposed in this Pull Request

- Add feature flag: `jetpack/user-licensing-m1`
- Enabled by default in `development` and `horizon` environments.
- Disabled by default in **all other** environments.

Asana task: 1200738571375997-as-1200794056860711

#### Testing instructions

1. Checkout and run this PR (`yarn start`).
1. Go to: http://calypso.localhost:3000/checkout/jetpack/thank-you/no-site/jetpack_security_daily?receiptId=9999&siteId=9999
1. Verify you see the text: `jetpack/user-licensing-m1: is enabled!` at the top of the page, just above the main page heading. (see screenshot)
1. In Staging environment (proxied), go to: https://wordpress.com/checkout/jetpack/thank-you/no-site/jetpack_security_daily?receiptId=9999&siteId=9999
1. Verify you **do not** see the text: `jetpack/user-licensing-m1: is enabled!`
1. In Production environment (Turn proxy off), go to: https://wordpress.com/checkout/jetpack/thank-you/no-site/jetpack_security_daily?receiptId=9999&siteId=9999
1. Verify you **do not** see the text: `jetpack/user-licensing-m1: is enabled!`

#### Screenshot

Markup 2021-08-30 at 09 10 54 (1)

*

### CODE REVIEW

**Review by rcanepa — COMMENTED** [review]
The name looks fine to me and the flag works as expected. I noticed that you enabled it in development and horizon, but not in stage. Is there a reason for that?

I think we can merge this once we remove the code that confirms the feature flag is enabled.

**Comment by elliottprogrammer** [comment]
>  I noticed that you enabled it in development and horizon, but not in stage. Is there a reason for that?

Oh, I thought we would only want the feature flag code available to for testing in the `development` environment and also in `horizon` so we could also test our features by clicking the calypso.live link.  I was thinking we would not **yet** want our changes available to all A18N's in staging and that we would enable in staging later in the future when milestone 1 was ready for an internal call for testing. But would you prefer to go ahead and enable the flag for staging now?

> I think we can merge this once we remove the code that confirms the feature flag is enabled.

Also my thoughts here were: that since the text confirming that the flag is enabled is only available in `development` and `horizon`, we could/would just remove the the code that displays the text confirmation on the next PR after this.  But yes, maybe it would be better remove the confirmation text in this PR, just before merging. That works too. 🙂

**Comment by rcanepa** [comment]
> Oh, I thought we would only want the feature flag code available to for testing in the development environment and also in horizon so we could also test our features by clicking the calypso.live link. I was thinking we would not yet want our changes available to all A18N's in staging and that we would enable in staging later in the future when milestone 1 was ready for an internal call for testing. But would you prefer to go ahead and enable the flag for staging now?

Having it disabled in staging, for now, sounds reasonable to me. Thanks for the explanation.

> Also my thoughts here were: that since the text confirming that the flag is enabled is only available in development and horizon, we could/would just remove the the code that displays the text confirmation on the next PR after this. But yes, maybe it would be better remove the confirmation text in this PR, just before merging. That works too. 🙂

I don't think it's necessary. We add feature flags all the time and we can trust that adding them on the right config file will make them work.

**Comment by elliottprogrammer** [comment]
> I don't think it's necessary. We add feature flags all the time and we can trust that adding them on the right config file will make them work.

Ok yep! Makes good sense, no problem! I'll remove the "testing text" right away right now so we can ship this. 👍

**Review by rcanepa — APPROVED** [review]
LGTM!

**Pull request merged**
